File: src/Memories/Domain/MemoryRepositoryInterface.php
<?php

namespace App\Memories\Domain;

use App\Shared\Domain\Uuid;

interface MemoryRepositoryInterface
{
    public function findById(Uuid $id): ?Memory;
    public function findByCriteria(array $criterias): array;

    /**
     * @return Memory[]
     */
    public function findAll(): array;
    public function save(Memory $memory): void;
    public function delete(Uuid $id): void;
    public function count(): int;
}

File: src/main/kotlin/com/zaleort/second_brain/Memories/Domain/Memory/MemoryRepositoryInterface.kt
package com.zaleort.second_brain.Memories.Domain.Memory

import com.zaleort.second_brain.Shared.Domain.QueryFilter.PaginatedResult
import com.zaleort.second_brain.Users.Domain.User.UserId

interface MemoryRepositoryInterface {
    fun findById(id: MemoryId): Memory?
    fun findByUserId(userId: UserId): List<Memory>
    fun save(memory: Memory)
    fun deleteById(id: MemoryId)
    fun search(params: MemorySearchParams): PaginatedResult<Memory>
}